### PR TITLE
Improve completion

### PR DESCRIPTION
**Pull request by catull**
Added missing arguments and option completions.
Alphabetised the options and option variables.

The argument "i" is not an alias for init, but rather for install.

If the argument is any one of "completions", "i" or "install", no suggestion will be displayed.
These arguments have no further options.

### CODE REVIEW

**Comment by catull** [comment]
I signed it!

File: addon/ng2/utilities/completion.sh [hunk 1]
```diff
@@ -1,18 +1,24 @@
...
-            --live-reload-port --environment --output-path --ssl --ssl-key --ssl-cert'
-generate_opts='component directive pipe route service'
-test_opts='--watch --browsers --colors --log-level --port --reporters'
+ng_opts='addon asset-sizes b build completion d destroy doc e2e g generate get h help i init install lint make-this-awesome new s serve server set t test v version -h --help'
+
+addon_opts='-b --blueprint -d -dir --directory --dry-run -sb --skip-bower -sg --skip-git -sn --skip-npm -v --verbose'
+asset_sizes_opts='-o --output-path'
+build_opts='--environment --output-path --suppress-sizes --watch --watcher -dev -e -prod'
+destroy_opts='--dry-run --verbose --pod --classic --dummy --in-repo --in-repo-addon -d -v -p -c -dum -id -i'
+generate_opts='component directive pipe route service --generate -d --dry-run --verbose -v --pod -p --classic -c --dummy -dum -id --in-repo --in-repo-addon -ir'
```
Comment (esinek) [inline addon/ng2/utilities/completion.sh]: don't forget about generating these types too: `class`, `enum`, `interface`

Comment (catull) [inline addon/ng2/utilities/completion.sh]: These are not mentioned when calling

```sh
ng help
```
That is the base for my change.

**Comment by catull** [comment]
Okay, extended the options by matching them up with the code.

File: addon/ng2/utilities/completion.sh [hunk 2]
```diff
@@ -1,18 +1,25 @@
...
-init_opts='--dry-run --verbose --blueprint --skip-npm --skip-bower --name'
-new_opts='--dry-run --verbose --blueprint --skip-npm --skip-bower --skip-git --directory'
-build_opts='--environment --output-path --watch --watcher'
-serve_opts='--port --host --proxy --insecure-proxy --watcher --live-reload --live-reload-host
-            --live-reload-port --environment --output-path --ssl --ssl-key --ssl-cert'
-generate_opts='component directive pipe route service'
-test_opts='--watch --browsers --colors --log-level --port --reporters'
+ng_opts='addon asset-sizes b build completion d destroy doc e2e g generate get github-pages:deploy gh-pages:deploy h help i init install lint make-this-awesome new s serve server set t test v version -h --help'
+
+addon_opts='-b --blueprint -d -dir --directory --dry-run -sb --skip-bower -sg --skip-git -sn --skip-npm -v --verbose'
```
Comment (filipesilva) [inline addon/ng2/utilities/completion.sh]: We don't really support addons at the moment in Angular-CLI itself, it's a remnant of the ember-cli functionality. Can you remove completion for these?

File: addon/ng2/utilities/completion.sh [hunk 3]
```diff
@@ -1,18 +1,25 @@
...
 
-ng_opts='new init build serve generate autocomplete e2e lint test version'
-init_opts='--dry-run --verbose --blueprint --skip-npm --skip-bower --name'
-new_opts='--dry-run --verbose --blueprint --skip-npm --skip-bower --skip-git --directory'
-build_opts='--environment --output-path --watch --watcher'
-serve_opts='--port --host --proxy --insecure-proxy --watcher --live-reload --live-reload-host
-            --live-reload-port --environment --output-path --ssl --ssl-key --ssl-cert'
-generate_opts='component directive pipe route service'
-test_opts='--watch --browsers --colors --log-level --port --reporters'
+ng_opts='addon asset-sizes b build completion d destroy doc e2e g generate get github-pages:deploy gh-pages:deploy h help i init install lint make-this-awesome new s serve server set t test v version -h --help'
```
Comment (filipesilva) [inline addon/ng2/utilities/completion.sh]: `asset-sizes d destroy`, we do not support these commands currently. They might work, somewhat, but we do not support them.

File: addon/ng2/utilities/completion.sh [hunk 4]
```diff
@@ -1,18 +1,25 @@
...
+build_opts='--environment --output-path --suppress-sizes --target --watch --watcher -dev -e -prod'
+destroy_opts='--dry-run --verbose --pod --classic --dummy --in-repo --in-repo-addon -d -v -p -c -dum -id -i'
+generate_opts='class component directive enum module pipe route service --generate -d --dry-run --verbose -v --pod -p --classic -c --dummy -dum -id --in-repo --in-repo-addon -ir'
+github_pages_deploy_opts='--environment --gh-token --gh-username --skip-build --user-page --message'
+help_opts='--json --verbose -v'
+init_opts='--blueprint --dry-run --link-cli --mobile --name --prefix --skip-bower --skip-npm --source-dir --style --verbose -b -d -lc -n -p -sb -sd -sn -v'
+new_opts='--blueprint --directory --dry-run --link-cli --mobile --prefix --skip-bower --skip-git --skip-npm --source-dir --style --verbose -b -d -dir -lc -p -sb -sd -sg -sn -v'
+serve_opts='--environment --host --insecure-proxy --inspr --live-reload --live-reload-base-url --live-reload-host --live-reload-live-css --live-reload-port --output-path --port --proxy --ssl --ssl-cert --ssl-key --target --watcher -H -dev -e -lr -lrbu -lrh -lrp -op -out -p -pr -prod -pxy -t -w'
+set_opts='==global -g'
```
Comment (filipesilva) [inline addon/ng2/utilities/completion.sh]: I think this should be `--global` instead.

File: addon/ng2/utilities/completion.sh [hunk 5]
```diff
@@ -1,18 +1,25 @@
...
+test_opts='--browsers --colors --config-file --environment --filter --host --launch --log-level --module --path --port --query --reporter --server --silent --test-page --test-port --watch -H -c -cf -e -f -m -r -s -tp -w'
```
Comment (filipesilva) [inline addon/ng2/utilities/completion.sh]: I don't think we support most of these to be honest.

**Comment by filipesilva** [comment]
Heya, thanks for making this PR! Some of the options you added completion to are things we don't actually support... which is hard to know without checking `addon/ng2/commands` per se. 

Can you update the PR to reflect the options there?

**Comment by catull** [comment]
In short, I have to remove `addon`, `asset-sizes`, `d` and `destroy`.
`set`'s option ==global must be --global.

I scanned all the source files in addon/ng2/commands.
I collected all `availableOptions`.
My assumption is if the command  mentions an option under its `availableOptions`, that means it is used, and thus must be completed.

**Comment by filipesilva** [comment]
@catull your assumption is correct, and thank you for the speedy update.

**Comment by filipesilva** [comment]
This will be merged soon. Don't worry too much about merging master in, I'll rebase when merging a few PRs together.

**Pull request closed without merging**